language/libra-vm/src/libra_vm.rs

```diff
@@ -470,20 +466,34 @@ impl LibraVM {
         result
     }
 
+    fn load_writeset(
```

Comment:
I find this name confusing. It sounds like you are loading the write set from the storage layer, whereas what you're actually doing is (IIUC) reading each access path that the write set is going to update.

I think it would be clearer to inline the function and expand the comment  to clarify (e.g., explain that all Move executions satisfy the read-before-write property).